### PR TITLE
audiality2, koboredux: init

### PR DESCRIPTION
**Pull request by fgaz**
###### Motivation for this change

Great game similar to the already packaged kobodeluxe. I think packaging also the proprietary part is fair, since in doing so we encourage a sensible foss business model. If you are a nixpkgs committer and want to test the proprietary assets, you can contact me and I'll send you a key.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fgaz** [comment]
Hmm why no build? Is it because of the pr's title?

@GrahamcOfBorg build koboredux

**Review by timokau — APPROVED** [review]
Very neat! Everything looks good to me. I think I can trust you that you have tested the proprietary assets, though the offer is appreciated :)

One thing to consider: We could also default `useProprietaryAssets` to `true`. This would

- make sure the users are getting the experience the developer intended by default
- further encourage the business model, which indeed looks very fair

We could then add another attribute, `koboredux-free`, that disables the proprietary assets. It could also override `meta.description` appropriately to show the relation. The error message of `requireFile` could point people to the free version.

The biggest downside is that the unfree license could prevent people from even giving it a shot and seeing the error message. Those people would probably find the `-free` version the same way they would have found the unfree version though.

What do you think? I don't feel very strongly about this either way. If you prefer it the way it is now, I'll merge.

**Comment by fgaz** [comment]
@timokau that's a great idea, I'll change it.

> The biggest downside is that the unfree license could prevent people from even giving it a shot and seeing the error message

Yes, this is a shame, I wish we had an exception or a special license attribute for this case. If the descriptions reference each other it isn't so bad though

**Comment by timokau** [comment]
Yeah, I doubt many people would be bothered by unfree assets in practice (especially if there are free substitutes).

Great, ping me when this is ready for another review.

**Comment by fgaz** [comment]
(busy times, I'll pick this up again sometimes this month)

/marvin opt-in
/status needs_work

**Comment by fgaz** [comment]
Well, that was a long time for a couple of string changes :sweat_smile: 

@timokau how about this? too pushy?

File: pkgs/games/koboredux/default.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  buildInputs = [
+    SDL2
+    SDL2_image
+    audiality2
+  ];
+
+  meta = {
+    description = "A frantic 80's style 2D shooter, inspired by the look and feel of 90's arcade cabinets" +
+      optionalString (!useProprietaryAssets)
+        " (This version replaces the official proprietary assets with placeholders. Consider installing \"koboredux\" instead)";
```
Comment (timokau) [inline pkgs/games/koboredux/default.nix:71]: This is too long for `description`. Consider moving it to `longDescription` and maybe add a succinct "community-edition" or similar to the regular description.

**Review by timokau — CHANGES_REQUESTED** [review]
Looks good, except the overly-long description. Sorry for the delay, I had some open-source holidays.

**Review by glittershark — APPROVED** [review]
Code LGTM besides the existing comment about the too-long description. Happy to test if necessary, but it seems like that's already been taken care of?

File: pkgs/games/koboredux/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/games/koboredux/default.nix]: Reworked the description

File: pkgs/games/koboredux/default.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  ];
+
+  buildInputs = [
+    SDL2
+    SDL2_image
+    audiality2
+  ];
+
+  meta = {
+    description = "A frantic 80's style 2D shooter, similar to XKobo and Kobo Deluxe" +
```
Comment (timokau) [inline pkgs/games/koboredux/default.nix:69]: This is missing the semi-colon at the end. Also the short description [should](https://nixos.org/manual/nixpkgs/stable/#sec-standard-meta-attributes) end with a period.

**Review by timokau — CHANGES_REQUESTED** [review]
Looks good otherwise.

File: pkgs/games/koboredux/default.nix (hunk 2 above)
Comment (fgaz) [reply pkgs/games/koboredux/default.nix:69]: Fixed the semicolon.

From your link:

> **Don’t** include a period at the end

Comment (timokau) [reply pkgs/games/koboredux/default.nix:69]: :laughing: let's say I was testing if you actually looked at the linked docs (I must have gotten confused with the guidelines for python docstrings and didn't bother to re-read the reference I was linking).

**Review by timokau — APPROVED** [review]

**Comment by timokau** [comment]
Thanks!

File: pkgs/games/koboredux/default.nix (hunk 2 above)
Comment (fgaz) [reply pkgs/games/koboredux/default.nix:69]: :)

**Pull request merged**